src/utils/tree_node.C

```diff
@@ -485,40 +493,33 @@ const Elem* TreeNode<N>::find_element(const Point& p, const std::set<subdomain_i
 
 
 template <unsigned int N>
-const Elem* TreeNode<N>::find_element_in_children(const Point& p, const std::set<subdomain_id_type> *allowed_subdomains) const
+const Elem* TreeNode<N>::find_element_in_children
+(const Point& p,
+ const std::set<subdomain_id_type> *allowed_subdomains,
+ Real relative_tol) const
 {
   libmesh_assert (!this->active());
 
-  unsigned int excluded_child = libMesh::invalid_uint;
+  std::vector<bool> searched_child(children.size(), false);
```

Comment:
It was the "write on one thread, read nearby element on another" case like you were saying.

All write's _and_ reads to `std::vector<bool>` must be blocking in threaded code... even if two processors are working on completely disjoint sets of entries...

All of that said: this is a _local_ variable... so there is no chance of it having threading issues.  Not only that but we need it to be fairly small and quickly searchable because it is in a critical code path.  I would vote for keeping it as `std::vector<bool>`